src/profiling/meter_manager.cpp

```diff
@@ -9,22 +9,23 @@ meter_manager::meter_manager() {
     meters.emplace_back(new time_meter());
 
     // add memory consumption meter
-    // TODO
+    if (has_memory_metering) {
+        meters.emplace_back(new memory_meter());
+    }
 
     // add energy consumption meter
     // TODO
 };
 
 void meter_manager::checkpoint(std::string name) {
-    checkpoint_names.push_back(std::move(name));
-
     // Enforce a global synchronization point the first time that the meters
     // are used, to ensure that times measured across all domains are
     // synchronised.
-    if (meters.size()==0) {
+    if (checkpoint_names.size()==0) {
```

Comment:
Correct behaviour of this method relies upon the public state of the struct members (i.e. `checkpoint_names`) having not been modified. If `checkpoint_names` remains public, then a private initialized flag of some sort should be added to keep this safe.